lib/opentelemetry/trace/recording_span.lua

```diff
@@ -135,9 +135,53 @@ function _M.tracer_provider(self)
     return self.tracer.provider
 end
 
+local function hex2bytes(str)
+    return (str:gsub('..', function(cc)
+        local n = tonumber(cc, 16)
+        if n then
+            return string.char(n)
+        end
+    end))
+end
+
+function _M.for_export(self)
+    return {
+        trace_id = self.ctx.trace_id,
+        span_id = self.ctx.span_id,
+        trace_state = "",
+        parent_span_id = self.parent_ctx.span_id or "",
+        name = self.name,
+        kind = self.kind,
+        start_time_unix_nano = string.format("%d", self.start_time),
+        end_time_unix_nano = string.format("%d", self.end_time),
+        attributes = self.attributes,
+        dropped_attributes_count = 0,
+        events = self.events,
+        dropped_events_count = 0,
+        links = {},
+        dropped_links_count = 0,
+        status = self.status
+    }
+end
+
+function _M.for_otlp_export(self)
+    local ret = self:for_export()
+    ret.trace_id = hex2bytes(ret.trace_id)
+    ret.span_id = hex2bytes(ret.span_id)
+    ret.parent_span_id = hex2bytes(ret.parent_span_id)
+    return ret
+end
+
+function _M.for_console_export(self)
+    local ret = "\n---------------------------------------------------------\n"
+    ret = ret .. util.table_as_string(self:for_export(), 2)
+    ret = ret .. "---------------------------------------------------------\n"
+    return ret
+end
```

Comment:
`for_otlp_export` and `for_console_export` is about export span logic, I think we should encapsulate in `lib/opentelemetry/trace/exporter/encoder.lua`. Then `encoder.to_oltp(span)` for oltp exporter, `encoder.to_console(span)` for console exporter.

Reply:
Great idea - will do.

Reply:
326750a75994b9c7b44f10dd0e573160d9b49022